numpy/_core/tests/test_dtype.py

```diff
@@ -991,8 +991,12 @@ def test_tuple_recursion(self):
         d = np.int32
         for i in range(100000):
             d = (d, (1,))
-        with pytest.raises(RecursionError):
+        # depending on OS and Python version, this might succeed
+        # see gh-30370 and cpython issue #142253
+        try:
             np.dtype(d)
+        except RecursionError:
+            pass
```

Comment:
nit: [`contextlib.suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) could also be used here